src/webchat-ui/components/presentational/RatingDialog.tsx

```diff
@@ -0,0 +1,158 @@
+    alignItems: "center",
+    flexGrow: 1,
+}));
+
+const RatingIconButton = styled(IconButton)(({ theme, selected }) => ({
+    backgroundColor: selected ? theme.primaryColor : theme.greyWeakColor,
+    borderRadius: "50%",
+
+    "&:focus, &:hover": {
+        backgroundColor: selected ? theme.primaryWeakColor : theme.greyColor,
+        opacity: 0.85,
+    },
+
+    'svg': {
+        width: 33,
+        height: 33,
+    },
+}));
+
+const RatingToolbar = styled(Toolbar)(({ theme }) => ({
+    paddingBottom: theme.unitSize,
+}));
+
+const RatingInput = styled(Textarea)(({ theme }) => ({
+    borderColor: theme.primaryWeakColor,
+}));
+
+interface IRatingDialogProps {
+    onCloseRatingDialog: () => void;
+    ratingValue: number | null;
+    ratingText: string;
+    onSetRatingText: ChangeEventHandler<HTMLInputElement>;
+    onSetPositiveRating: () => void;
+    onSetNegativeRating: () => void;
+    onSendRating: () => void;
+    disableSendButton: boolean;
+    ratingTitleText: string;
+    ratingCommentText: string;
+}
+
+export default ({
+    onCloseRatingDialog,
+    ratingValue,
+    ratingText,
+    onSetRatingText,
+    onSetPositiveRating,
+    onSetNegativeRating,
+    onSendRating,
+    disableSendButton,
+    ratingTitleText,
+    ratingCommentText,
+    ...props
+}: IRatingDialogProps) => (
+    <Wrapper>
+        <RatingDialogRoot>
+            <RatingDialogHeader>
+                <span>{ratingTitleText}</span>
+                <IconButton onClick={onCloseRatingDialog}>
+                    <CloseIcon />
+                </IconButton>
+            </RatingDialogHeader>
+            <RatingDialogMain>
+                <RatingButtonContainer>
+                    <div>
+                        <RatingIconButton onClick={onSetPositiveRating} selected={ratingValue === 10}>
+                            <ThumbIcon />
+                        </RatingIconButton>
+                    </div>
+                    <div>
+                        <RatingIconButton onClick={onSetNegativeRating} selected={ratingValue === 0}>
+                            <ThumbDownIcon />
+                        </RatingIconButton>
+                    </div>
+                </RatingButtonContainer>
+                <div>
+                    <div>{ratingCommentText}</div>
+                </div>
+            </RatingDialogMain>
+            <RatingToolbar>
+                <RatingInput
+                    type="text"
+                    value={ratingText}
+                    onChange={onSetRatingText}
+                    autoFocus
+                    maxlength={500}
+                    rows={3}
+                />
+                <IconButton
+                    className={disableSendButton ? "disabled" : "active"}
+                    disabled={disableSendButton}
+                    onClick={onSendRating}
+                >
+                    <SendIcon />
+                </IconButton>
+            </RatingToolbar>
```

Comment:
Well of course, that's what I wanted to do. I only put it in the store since you asked for that explicitly two days ago :)

Reply:
That was more about the "result" and less about the "intermediate state of the dialog"

Reply:
Since the given rating is part of the message history, I don't see a reason to store it in the store. Do you agree?

Reply:
That's true, I'm just not sure whether it would be too expensive to derive it from there. You would need to find the last rating message in the chat history and then check its contents. If we can use a selector for that or memoize it in other ways it would be fine i guess.

Reply:
We're not deriving it from the message history. That's why I added the "ratingGiven" store variable (to be renamed). It is persisted and it's purpose is to remember that a rating was already given in this session.

Reply:
refactored to use local state